### ISSUE TITLE
MayaCharm 3.2.7 使用时发生错误！

### ISSUE THREAD

**Issue by riggingWang** [cmcpasserby/MayaCharm#70]
错误信息： ’encoding’ is an invalid keyword argument for this function

**Comment by cmcpasserby** [comment]
looks like bug from last release, are you using python 2 or 3 think this might only happen with 2

**Comment by riggingWang** [comment]
I used a Python2 , When I switched back to the previous version, the error disappeared.

**Comment by cmcpasserby** [comment]
Yeah i will have to look at fixing that on the weekend, will either need to detect what mode maya is in, or provide a per project setting for python 2 or 3 support.
Was #69 that caused this bug to get in.

**Comment by riggingWang** [comment]
This plugin is very useful, thank you for your efforts!

**Comment by 44hero** [comment]
maya2022's py2 and py3 systems, both
Getting MayaCharm to work properly is more difficult than I expected.
Thank you very much for always responding well.

Thank you for your continued support.

far from Japan

**Comment by 44hero** [comment]
If you create a program on the assumption that the file to be read by the open() function is UTF-8, it seems that you must add encoding='utf-8' so that it will be read in UTF-8 even in a Windows environment.
Maybe it's because in English speaking countries, only characters in the ASCII range are used, and even if UTF-8 and Windows-1252 have different character codes, they don't cause many errors.
There are quite a lot of libraries that forget to do this, and it seems to happen quite often.

For example, the moduleFinder library is one of them.

Therefore, unlike my MayaCharm, you cannot modify the contents of the moduleFinder library.
For the py3 system, I have added the following code, and the current situation is surpassed. . . 😅
It's a pain. .

# Workaround to throw an error Description ###################################### start
# ########################################
# explanation
# ########################################
# The python3-based code may generate the following error for the python standard library modulefinder.
# This is a description to avoid that. Common error samples include the following:
# ####################
# Error sample #######
# ####################
# modulefinder.py line 287: 'cp932' codec can't decode byte 0x87 in position 264: illegal multibyte sequence
import_locale
_locale._getdefaultlocale = (lambda *args: ['en_US', 'utf8'])
# Workaround to emit error Description ###################################### end

**Comment by 44hero** [comment]
If you create a program on the assumption that the file to be read by the open() function is UTF-8, it seems that you must add encoding='utf-8' so that it will be read in UTF-8 even in a Windows environment.
Maybe it's because in English speaking countries, only characters in the ASCII range are used, and even if UTF-8 and Windows-1252 have different character codes, they don't cause many errors.
There are quite a lot of libraries that forget to do this, and it seems to happen quite often.

For example, the moduleFinder library is one of them.

Therefore, unlike my MayaCharm, you cannot modify the contents of the moduleFinder library.
For the py3 system, I have added the following code, and the current situation is surpassed. . . 😅
It's a pain. .

```python=
# Workaround to throw an error Description ###################################### start
# ########################################
# explanation
# ########################################
# The python3-based code may generate the following error for the python standard library modulefinder.
# This is a description to avoid that. Common error samples include the following:
# ####################
# Error sample #######
# ####################
# modulefinder.py line 287: 'cp932' codec can't decode byte 0x87 in position 264: illegal multibyte sequence
import_locale
_locale._getdefaultlocale = (lambda *args: ['en_US', 'utf8'])
# Workaround to emit error Description ###################################### end
```

**Comment by 0lxf0** [comment]
> 错误信息： ’encoding’ is an invalid keyword argument for this function错误信息： 'encoding' is an invalid keyword argument for this function

waitting for new version
